crates/pop-parachains/src/build.rs

```diff
@@ -1,13 +1,498 @@
+///
+/// # Arguments
+/// * `target_path` - The path where the binaries are expected to be found.
+/// * `node_path` - The path to the node from which the node name will be parsed.
+pub fn binary_path(target_path: &Path, node_path: &Path) -> Result<PathBuf, Error> {
+	let node_name = parse_node_name(node_path)?;
+	let release = target_path.join(node_name.clone());
+	if !release.exists() {
+		return Err(Error::MissingBinary(node_name));
+	}
+	Ok(release)
+}
+
+/// Generates the plain text chain specification for a parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `binary_path` - The path to the node binary executable that contains the `build-spec` command.
+/// * `chain_spec_file_name` - The name of the chain specification file to be generated.
+/// * `para_id` - The parachain ID to be replaced in the specification.
+pub fn generate_chain_spec(
+	path: Option<&Path>,
+	binary_path: &Path,
+	chain_spec_file_name: &str,
+	para_id: u32,
+) -> Result<PathBuf, Error> {
+	check_command_exists(&binary_path, "build-spec")?;
+	let plain_parachain_spec = path.unwrap_or(Path::new("./")).join(chain_spec_file_name);
+	cmd(binary_path, vec!["build-spec", "--disable-default-bootnode"])
+		.stdout_path(plain_parachain_spec.clone())
+		.run()?;
+	let generated_para_id = get_parachain_id(&plain_parachain_spec)?;
+	replace_para_id(plain_parachain_spec.clone(), para_id, generated_para_id)?;
+	Ok(plain_parachain_spec)
+}
+
+/// Generates a raw chain specification file for a parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the plain chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `build-spec` command.
+/// * `chain_spec_file_name` - The name of the chain specification file to be generated.
+pub fn generate_raw_chain_spec(
+	path: Option<&Path>,
+	plain_chain_spec: &Path,
+	binary_path: &Path,
+	chain_spec_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !plain_chain_spec.exists() {
+		return Err(Error::MissingChainSpec(plain_chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "build-spec")?;
+	let raw_chain_spec = path.unwrap_or(Path::new("./")).join(chain_spec_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"build-spec",
+			"--chain",
+			&plain_chain_spec.display().to_string(),
+			"--disable-default-bootnode",
+			"--raw",
+		],
+	)
+	.stdout_path(raw_chain_spec.clone())
+	.run()?;
+	Ok(raw_chain_spec)
+}
+
+/// Export the WebAssembly runtime for the parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-wasm` command.
+/// * `wasm_file_name` - The name of the wasm runtime file to be generated.
+pub fn export_wasm_file(
+	path: Option<&Path>,
```

Comment:
Again, either take a path for the wasm file, or use the chain_spec location to output in the same directory.

Please also consider the ordering of these parameters. Logically to me I would specify the binary first, as that is used to generate the wasm blob, then the chain spec file, then resulting wasm path. i.e. the same order as those passed to the cmd function within. Same applies for the other functions.

Reply:
This order makes more sense.